File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        
        ListNode start = new ListNode(0);
        ListNode sp=start, fp=start;
        start.next=head;
        
        for(int i=1; i<=n+1; i++)
        fp = fp.next;
    
        while(fp!=null){
            fp=fp.next;
            sp=sp.next;          
        }        
        sp.next=sp.next.next;
        return start.next;        
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        fp=sp= head
        for _ in range(n):
            fp=fp.next
        if not fp: return head.next #for single [1]
        while fp.next:
            fp=fp.next
            sp=sp.next
        sp.next=sp.next.next
        return head
    
             
            
        